Sources/AccountSDKIOSWeb/Simplified Login/WebViewController.swift

```diff
@@ -1,19 +1,91 @@
 import WebKit
 
-class WebViewController: UIViewController, WKNavigationDelegate {
-
-    let webView: WKWebView
+class WebViewController: UIViewController {
+    
+    private let webView: WKWebView
+    private let progressView = UIProgressView(progressViewStyle: .default)
+    private let activityIndicatorView = UIActivityIndicatorView()
+    
+    private var estimatedProgressObserver: NSKeyValueObservation?
+    
+    init() {
+        webView = WKWebView()
+        webView.backgroundColor = .white
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
     
     override func loadView() {
         webView.navigationDelegate = self
         view = webView
+        
+        addLoadingViews()
+        setupEstimatedProgressObserver()
     }
     
-    init(url: URL) {
-        webView = WKWebView()
-        webView.load(URLRequest(url: url))
-        super.init(nibName: nil, bundle: nil)
+    func loadURL(_ url: URL) {
+        DispatchQueue.main.async {
+            self.webView.load(URLRequest(url: url))
+        }
+    }
+    
+    private func addLoadingViews() {
+        guard let navigationBar = navigationController?.navigationBar else {
+            return
```

Comment:
Will ths mean If no navigation controller exists. Then no loading indicator will be shown, not even the spinner?

Reply:
There will be a spinner without progress bar :)

Reply:
This guard will require `WebViewController` to be embedded inside a navigation controller, to run the rest of the function logic. 

If that is the intention for the progress bar and not the spinner then please modify the guard to cover only the progress bar logic.

Reply:
Oh yes, i missed it 🤦‍♀️ Thank you 🙌